terralab/custom_http_server.py

```diff
@@ -0,0 +1,102 @@
+"""
+Custom local HTTP server logic used for OAuth2 flow
+"""
+
+from http.server import HTTPServer
+from typing import Optional
+from urllib.parse import parse_qs
+
+from oauth2_cli_auth.http_server import OAuthRedirectHandler
+from oauth2_cli_auth._timeout import TimeoutException, _method_with_timeout
+
+
+class CustomOAuthRedirectHandler(OAuthRedirectHandler):
+    """
+    HTTPRequest Handler that is intended to be used as oauth2 callback page.
+    Customized for terralab to receive and parse a POST request from the redirect.
+    """
+
+    def do_GET(self) -> None:
+        """For security reasons, we don't ever want to process a GET request."""
+        return NotImplementedError("GET request not supported for authentication flow.")
```

Comment:
`NotImplemented` is supposed to be used for binary special methods, see https://docs.python.org/3/library/constants.html#NotImplemented